spec/stack_master/commands/status_spec.rb

```diff
@@ -35,10 +41,66 @@
       let(:proposed_stack2) { double(:proposed_stack2, template_body: "{}", template_format: :json, parameters_with_defaults: {a: 1}) }
 
       it "returns the status of call stacks" do
-        out = "REGION    | STACK_NAME | STACK_STATUS    | DIFFERENT\n----------|------------|-----------------|----------\nus-east-1 | stack1     | UPDATE_COMPLETE | Yes      \nus-east-1 | stack2     | CREATE_COMPLETE | No       \n * No echo parameters can't be diffed\n"
+        out = <<~OUTPUT
+          REGION    | STACK_NAME | STACK_STATUS    | DIFFERENT
+          ----------|------------|-----------------|----------
+          us-east-1 | stack1     | UPDATE_COMPLETE | Yes      
+          us-east-1 | stack2     | CREATE_COMPLETE | No       
+           * No echo parameters can't be diffed
+        OUTPUT
         expect { status.perform }.to output(out).to_stdout
       end
     end
+
+    context 'when identity account is not allowed' do
+      let(:sts) { Aws::STS::Client.new(stub_responses: true) }
+      let(:stack_definition_1) { double(:stack_definition_1, region: 'us-east-1', stack_name: 'stack1', allowed_accounts: ['not-account-id']) }
+      let(:stack1) { double(:stack1, template_body: '{"foo": "bar"}', template_hash: {foo: 'bar'}, template_format: :json, parameters_with_defaults: {a: 1}, stack_status: 'UPDATE_COMPLETE') }
+      let(:stack2) { double(:stack2, template_body: '{}', template_hash: {}, template_format: :json, parameters_with_defaults: {a: 1}, stack_status: 'CREATE_COMPLETE') }
+      let(:proposed_stack1) { double(:proposed_stack1, template_body: "{}", template_format: :json, parameters_with_defaults: {a: 1}) }
+      let(:proposed_stack2) { double(:proposed_stack2, template_body: "{}", template_format: :json, parameters_with_defaults: {a: 1}) }
+
+      before do
+        allow(Aws::STS::Client).to receive(:new).and_return(sts)
+        sts.stub_responses(:get_caller_identity, {
+          account: 'account-id',
+          arn: 'an-arn',
+          user_id: 'a-user-id'
+        })
+      end
+
+      it 'sets stack status and different fields accordingly' do
+        out = <<~OUTPUT
+          REGION    | STACK_NAME | STACK_STATUS       | DIFFERENT
+          ----------|------------|--------------------|----------
+          us-east-1 | stack1     | Disallowed account | N/A      
+          us-east-1 | stack2     | UPDATE_COMPLETE    | Yes      
+           * No echo parameters can't be diffed
+        OUTPUT
+        expect { status.perform }.to output(out).to_stdout
```

Comment:
Shouldn't this test expect an error to be returned? If the account IDs don't match it shouldn't print the status?

Reply:
It's a got a special case now: `Disallowed account | N/A`.
It would not be great if it threw an error just because one stack is unavailable.

Reply:
What @viraptor said.

Alternatively, I could leave them blank instead? I didn't go with that option because I thought it would be better to let the user know why no status is included for that stack.

Reply:
Another option is to include `SKIPPED` for the status field and `Disallowed account` for different field.

Reply:
Right I see. Are we able to have an integration test for when someone runs `stack_master apply` on a stack in the wrong account then? I thought that's what this was doing, but I didn't read the spec name :laughing:

Reply:
Good idea. Added integration tests in f02603c.